Invoke-DbaDbShrink: Adjusting the verbose messages to account for the usage of the dbasize type

lancasteradam opened the pull request:
<!-- Below information IS REQUIRED with every PR -->
## Type of Change
<!-- What type of change does your code introduce -->
 - [x] Bug fix (non-breaking change, relates to #7241  )
 - [ ] New feature (non-breaking change, adds functionality, fixes #<!--issue number--> )
 - [ ] Breaking change (effects multiple commands or functionality, fixes #<!--issue number--> )
 - [x] Ran manual Pester test and has passed (`.\tests\manual.pester.ps1`)
 - [ ] Adding code coverage to existing functionality
 - [ ] Pester test is included
 - [ ] If new file reference added for test, has is been added to github.com/dataplat/appveyor-lab ?
 - [ ] Unit test is included
 - [ ] Documentation
 - [ ] Build system

<!-- Below this line you can erase anything that is not applicable -->
### Purpose
<!-- What is the purpose or goal of this PR? (doesn't have to be an essay) -->
Fixing the verbose messages to display the correct sizes per the findings from @miguelmolgar and @andreasjordan.

### Approach
<!-- How does this change solve that purpose -->
Ensuring the verbose messages are multiplied by 1024 similar to how the result object is being done.

### Commands to test
<!-- if these are the examples in the help just note it as such -->
A new test was added.

### Sample output with this fix
VERBOSE: [08:31:59][Connect-DbaInstance] Server object passed in, will do some checks and then return the original object
VERBOSE: [08:31:59][Invoke-DbaDbShrink] Connection timeout set to 0
VERBOSE: [08:31:59][Invoke-DbaDbShrink] Processing [dbatoolsci_shrinktest] on []
VERBOSE: [08:31:59][Invoke-DbaDbShrink] File: dbatoolsci_shrinktest
**VERBOSE: [08:31:59][Invoke-DbaDbShrink] Initial Size: 16.00 MB
VERBOSE: [08:31:59][Invoke-DbaDbShrink] Space Used: 2.63 MB
VERBOSE: [08:31:59][Invoke-DbaDbShrink] Initial Freespace: 13.38 MB
VERBOSE: [08:31:59][Invoke-DbaDbShrink] Target Freespace: 0 B
VERBOSE: [08:31:59][Invoke-DbaDbShrink] Target FileSize: 2.63 MB**
VERBOSE: Performing the operation "Shrinking from 16.00 MB to 2.63 MB" on target "[dbatoolsci_shrinktest] on []".
VERBOSE: [08:31:59][Invoke-DbaDbShrink] Getting starting average fragmentation
VERBOSE: [08:31:59][Invoke-DbaDbShrink] Beginning shrink of files
VERBOSE: [08:31:59][Invoke-DbaDbShrink] ShrinkGap: 13.38 MB
VERBOSE: [08:31:59][Invoke-DbaDbShrink] Step Size: 2.00 MB
**VERBOSE: [08:31:59][Invoke-DbaDbShrink] Step: 1 of 6
VERBOSE: [08:31:59][Invoke-DbaDbShrink] Shrinking dbatoolsci_shrinktest to 14.00 MB
VERBOSE: [08:32:04][Invoke-DbaDbShrink] Step: 2 of 6
VERBOSE: [08:32:04][Invoke-DbaDbShrink] Shrinking dbatoolsci_shrinktest to 12.00 MB
VERBOSE: [08:32:04][Invoke-DbaDbShrink] Step: 3 of 6
VERBOSE: [08:32:04][Invoke-DbaDbShrink] Shrinking dbatoolsci_shrinktest to 10.00 MB
VERBOSE: [08:32:04][Invoke-DbaDbShrink] Step: 4 of 6
VERBOSE: [08:32:04][Invoke-DbaDbShrink] Shrinking dbatoolsci_shrinktest to 8.00 MB
VERBOSE: [08:32:05][Invoke-DbaDbShrink] Step: 5 of 6
VERBOSE: [08:32:05][Invoke-DbaDbShrink] Shrinking dbatoolsci_shrinktest to 6.00 MB
VERBOSE: [08:32:05][Invoke-DbaDbShrink] Step: 6 of 6
VERBOSE: [08:32:05][Invoke-DbaDbShrink] Shrinking dbatoolsci_shrinktest to 4.00 MB**
VERBOSE: [08:32:05][Invoke-DbaDbShrink] Final file size: 4.00 MB
VERBOSE: [08:32:05][Invoke-DbaDbShrink] Final file space available: 1.38 MB
VERBOSE: [08:32:05][Invoke-DbaDbShrink] Getting ending average fragmentation

andreasjordan commented:
You just "fix" the contents of the verbose message. The value of `$startingSize` is sill wrong. If you use the type [dbasize], you have to make sure that the value is in bytes. 
So please remove the use of [dbasize] and work in kilobytes - or use [dbasize] and implement the correct conversion when storing or using the values.

lancasteradam commented:
@andreasjordan, I added a new pester test to ensure that the file sizes are logged correctly with the verbose messages. The return object was already adjusting the values by 1024, so that is why I added the conversion for only the verbose messages. If you can describe a scenario that would cause the command to break I can write a new pester test to simulate it. Based on the output in the posting above I believe the command is working properly. However, I may be missing something so let me know if there is a different conversion method to be used with the dbasize datatype.

lancasteradam commented:
Thanks for those additional details. I'll take a look.
